### ISSUE TITLE
onLocation is not calling when kill the app android.

### ISSUE THREAD

**Issue by swapanil** [transistorsoft/react-native-background-geolocation#992]
## Your Environment
* Plugin version: ^3.4.2
* Platform: Android 
* OS version: 8 and 9
* Device manufacturer / model: Nokia/Samsung/Mi
* React Native version (`react-native -v`):  0.61.3
* Plugin config 

Ready:

>BackgroundGeolocation.ready({
      desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
      distanceFilter: 10,
      stopTimeout: 1,
      debug: true, 
      logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
      stopOnTerminate: false,  
      startOnBoot: true,  
      enableHeadless: true,      
      url: 'http://3.15.23.93:4000/users/background-location',
      batchSync: false,      
      autoSync: true,         
      headers: {              
        "X-FOO": "bar"
      },
      params: {             
        "auth_token": "maybe_your_server_authenticates_via_token_YES?"
      }
    }, (state) => {
      console.log("- BackgroundGeolocation is configured and ready: ", state.enabled);
      if (!state.enabled) {
        BackgroundGeolocation.start(function () {
          console.log("- Start success");
        });
      }
    });`

onLocation Method:

     onLocation(location) {
    console.log(location.coords)
    const lng = location.coords.longitude
    const lat = location.coords.latitude
    Axios.post(
      `${baseUrl}users/background-location`,
      [
        {
          lat: lat,
          lon: lng,
          deviceId: DeviceInfo.getUniqueId(),
          userId:
            store.getState().Secrets.secretsData &&
            store.getState().Secrets.secretsData._id
              ? store.getState().Secrets.secretsData._id
              : '', // you can also add your own properties
          from: 'Background Fetch 1',
        },
      ],
      {
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
      },
    ).then(function (response) {
      console.log('completed!')
      console.log(response);
    })
    .catch(function (error) {
      console.log(error);
    })
  }`
## Expected Behavior
Location should be updated to server through API on onLocation event.

## Actual Behavior
onLocation is not triggered if the app is closed.

## Steps to Reproduce
1.  Do some background stuff in onLocation
2. Kill the app.

## Context
Update location to the server whenever it gets changed.

## Debug logs
[https://swapanil.github.io/project/Terminal%20Saved%20Output](Logs)
<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
PASTE_YOUR_LOGS_HERE
```

</details>

**Comment by christocracy** [comment]
Of course not.  Your JavaScript app no longer exists when the Android app is terminated.

This is why the plugin has its own built-in SQLite database and http service.

See wiki “Android Headless Mode”.

**Comment by swapanil** [comment]
You mean onLogin will not be called after Terminating app, right? 

So the alternative I think the headless but i need to call an api as you can see. 

Can you tell me how can i do that using headless. 

Our client has purchased the license and now I have to complete this and make release apk. 

And there are two license key showing there one for less than version 3 and other for greater so i have chooses one with greater than three as of my library version I have mentioned. 

Can you please guide me how can i achieve this using headless. 

Basically, I have to upload location to server as you can see on posted issue. 

Looking forward for your reply. 

Thanks
